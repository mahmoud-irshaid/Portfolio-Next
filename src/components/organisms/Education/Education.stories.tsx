/* eslint-disable import/no-anonymous-default-export */
import { Education, EducationProps } from './Education';

import { mockSchool } from '@mockdata';

export default {
	title: 'organisms/Education',
	component: Education,
};

export const EducationStory = (args: EducationProps) => <Education {...args} />;

EducationStory.storyName = 'Education';
EducationStory.args = {
	education: [mockSchool],
};
