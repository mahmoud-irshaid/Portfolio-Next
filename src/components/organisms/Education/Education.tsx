import { School } from '@components/molecules/School';

import type { Education as EducationType } from '@types';
import classNames from 'classnames';

export interface EducationProps {
	education: EducationType[];
}

const Education = ({ education }: EducationProps) => {
	return (
		<>
			{education.map(({ course, date, institution, courseContents }) => (
				<div
					className={classNames(
						'flex transform transition-all md:hover:scale-[1.01]'
					)}
					key={course}
				>
					<School
						course={course}
						date={date}
						institution={institution}
						courseContents={courseContents}
					/>
				</div>
			))}
		</>
	);
};

export { Education };
