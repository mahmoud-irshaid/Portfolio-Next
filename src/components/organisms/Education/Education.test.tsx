import { Education } from '../Education';

import { mockSchool } from '@mockdata';
import { render } from '@testing-library/react';

describe('Education', () => {
	it('renders correctly', () => {
		const { container } = render(<Education education={[mockSchool]} />);
		expect(container).toMatchSnapshot();
	});
});
