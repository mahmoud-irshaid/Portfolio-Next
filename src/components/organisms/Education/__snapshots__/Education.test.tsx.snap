// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education renders correctly 1`] = `
<div>
  <div
    class="flex transform transition-all md:hover:scale-[1.01]"
  >
    <div
      class="w-full glass-light dark:glass-dark mb-6 rounded-xl px-8 py-6 drop-shadow-lg"
    >
      <div
        class="flex items-center gap-4"
      >
        <img
          alt="Some Company"
          class="rounded-md"
          data-nimg="1"
          decoding="async"
          height="64"
          loading="lazy"
          src="/assets/companies/dennis.svg"
          style="color: transparent;"
          width="64"
        />
        <div
          class="mb-2 w-full"
        >
          <div
            class="flex justify-between"
          >
            <h3
              class="font-bold"
            >
              Some course
            </h3>
            <span>
              Jan 2020
            </span>
          </div>
          <h4>
            <a
              class="underlined relative"
              href="https://somecompany.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Some Company
            </a>
          </h4>
          <p
            class="text-sm"
          >
            <strong>
              Course Contents: 
            </strong>
            TypeScript, React
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
